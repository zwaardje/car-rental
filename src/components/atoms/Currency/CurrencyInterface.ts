export interface Price {
    currency: string;
    amount: number;
}

export interface Currency {
    round: boolean;
    locales: string;
    price: Price;
}
