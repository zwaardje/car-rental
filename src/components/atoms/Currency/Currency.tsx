import * as React from 'react';

import { Currency as CurrencyInterface } from './CurrencyInterface';

export interface Props extends CurrencyInterface {}
export interface State {}

export default class Currency extends React.Component<Props, State> {
    static displayName = 'Currency';

    get formatRoundPrice ()  {
        const { locales, price: { currency, amount } } = this.props;

        return new Intl.NumberFormat(locales, {
            currency,
            style: 'currency',
            minimumFractionDigits: 2,
            maximumFractionDigits: 2,
        }).format(Math.round(amount));
    }

    get formatPrice () {
        const { locales, price: { currency, amount } } = this.props;
        return new Intl.NumberFormat(locales, {
            currency,
            style: 'currency',
        }).format(amount);
    }

    render() {
        const { round } = this.props;

        const amount = round ? this.formatRoundPrice : this.formatPrice;

        return (
            <span className="price">
                {amount}
            </span>
        );
    }
}
