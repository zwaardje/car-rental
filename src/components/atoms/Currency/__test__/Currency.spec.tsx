import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Currency from '../Currency';

import {
    defaultProps,
    round,
} from '../__data__/Currency.mockdata';

describe('Currency', () => {

    it('renders default dutch currency correctly', () => {
        const props = { ... defaultProps };
        const tree = renderer.create(<Currency {... props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders round dutch currency correctly', () => {
        const props = { ... round };
        const tree = renderer.create(<Currency {... props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
