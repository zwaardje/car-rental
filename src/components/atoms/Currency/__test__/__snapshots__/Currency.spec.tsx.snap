// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency renders default dutch currency correctly 1`] = `
<span
  className="price"
>
  € 43,39
</span>
`;

exports[`Currency renders round dutch currency correctly 1`] = `
<span
  className="price"
>
  € 43,00
</span>
`;
